### ISSUE TITLE
LoopingCall should parameterize its scheduler

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2633]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#2633|
|Type|enhancement|
|Created|2007-05-08 19:29:11Z|

_callLater and _seconds should both be provided or obsoleted by parameterized functions or a Clock/reactor-like object.

<details><summary>Searchable metadata</summary>

```
trac-id__2633 2633
type__enhancement enhancement
reporter__radix radix
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1178652551000000 1178652551000000
changetime__1179103111000000 1179103111000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_1"></a>|@radix removed owner|
|-|-|

Up for review at parameterize-loopingcall-scheduler-2633, which also attempts to resolve [#2634](https://github.com/twisted/twisted/issues/2634).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @radix|
|-|-|

 * I don't really like the setClock method. I mean, anybody can do ```loop.clock = foo```, right?
 * the variable new_time in t.i.base should be named newTime
 * the docstring of seconds in t.i.interfaces should be 
```
"""
blah.
"""
```
 Don't hesitate to clean the docstrings of IReactorTime :)
 * the test_seconds is a bit insane, but I guess you have a reason for that

Apart from nitpicking, that looks really great, so please merge when you're ok.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

One last thing: can you update the copyright notices of the touched files? Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_4"></a>|@radix set status to `closed`|
|-|-|

(In [20273]) Merge parameterize-looping-call-2633.

Author: radix
Reviewer: therve
Fixes [#2633](https://github.com/twisted/twisted/issues/2633)
Fixes [#2634](https://github.com/twisted/twisted/issues/2634)

LoopingCall (and code which uses it) is now much more testable because the
scheduler it uses is overridable. In support of this, a 'seconds' method
was added to the reactor (IReactorTime) to strengthen the logical grouping
of 'what is the current scheduling-time' and 'schedule a call for later'.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
